From 3420e9ac354235757862f685018a86e1315d662e Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Mon, 26 Sep 2022 10:00:51 -0300
Subject: [PATCH 120/142] linux: Avoid indirection on operand of type 'void *'
 for gettimeofday

ISO C does not allow and it fixes a clang issue with
-Werror,-Wvoid-ptr-dereference.

Checked on x86_64-linux-gnu.
---
 sysdeps/unix/sysv/linux/gettimeofday.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/gettimeofday.c b/sysdeps/unix/sysv/linux/gettimeofday.c
index 7ab147c614..11d6173ff1 100644
--- a/sysdeps/unix/sysv/linux/gettimeofday.c
+++ b/sysdeps/unix/sysv/linux/gettimeofday.c
@@ -32,7 +32,7 @@ static int
 __gettimeofday_syscall (struct timeval *restrict tv, void *restrict tz)
 {
   if (__glibc_unlikely (tz != 0))
-    memset (tz, 0, sizeof *tz);
+    memset (tz, 0, sizeof (struct timezone));
   return INLINE_SYSCALL_CALL (gettimeofday, tv, tz);
 }
 
@@ -48,7 +48,7 @@ int
 __gettimeofday (struct timeval *restrict tv, void *restrict tz)
 {
   if (__glibc_unlikely (tz != 0))
-    memset (tz, 0, sizeof *tz);
+    memset (tz, 0, sizeof (struct timezone));
 
   return INLINE_VSYSCALL (gettimeofday, 2, tv, tz);
 }
-- 
2.34.1

